src/main/java/org/logstash/dissect/Delim.java

```diff
@@ -0,0 +1,7 @@
+package org.logstash.dissect;
+
+interface Delim {
```

Comment:
In this case an implementation of Delim is two things 1) a holder for the delimiter bytes and 2) an algorithm best suited for searching for its byte(s) in a source byte array from a certain point onwards. I realise that I could plugin the strategy when building the Delim. Earlier versions of the code did have this. I had KMP and BoyerMoore algos but String.indexOf altered for bytes benchmarked quickest for the size of needle and haystack encountered here.

I will change this. I can plug in the strategy. I need one class called Delimiter in that case and the interface moves to SearchStrategy with one method: `int indexOf(byte[] data, int offset);`